### ISSUE TITLE
HTTP-triggered Function can be triggered despite being disabled

### ISSUE THREAD

**Issue by sebader** [Azure/azure-functions-host#4764]
According to [this ](https://docs.microsoft.com/en-us/azure/azure-functions/disable-function#functions-2x---all-languages)article I disabled a HTTP-triggered function using an app setting 

    "AzureWebJobs.Http2BlobFunction.Disabled": "true"

Starting up the function, it also shows up in the console output of the function runtime

    Function Http2BlobFunction is disabled

However, you can still call the function on its API and it gets executed alright. 
Looks like a bug to me.

#### Investigative information

Please provide the following:

Azure Functions Core Tools (2.7.1505 Commit hash: eb8182995562240ca83dd0e0e3394586cf5fdfa3)
Function Runtime Version: 2.0.12590.0

#### Repro steps

See above. Create an HTTP triggered function and disable it using an app setting.

#### Expected behavior

Function should not be executed when called. Return a 404 or something like this.

#### Actual behavior

Function does get executed.

![image](https://user-images.githubusercontent.com/11445087/62541022-35757100-b859-11e9-816a-bf542cad2576.png)
 
(The other two functions are not disabled, so that's ok. But Http2BlobFunction is disabled)

#### Known workarounds

None

#### Related information 

Only tested in C# so far.

**Comment by brettsam** [comment]
Does this only happen when running locally with the CLI (or VS)? Http requests should return 404s unless invoked via an Admin request -- this allows the portal to continue to be able to test functions. When running via the CLI, though, I believe all requests are treated as admin requests.

@ahmelsayed can correct me if I'm wrong.

**Comment by sebader** [comment]
@brettsam so far I only tested locally when starting debugging from VS

**Comment by sebader** [comment]
Yes, I can confirm this is indeed different when deployed in Azure. When I use the app setting there AND use the function key (not the master key), the call returns 404.

I would say to the very least the documentation page should state that - and ideally the console output in the run time should reflect this behavior as well.

**Comment by fabiocav** [comment]
Would be good to validate the behavior in Core Tools here. Transferring to the repo.
